primitives/vrf/src/lib.rs

```diff
@@ -0,0 +1,30 @@
+// Copyright 2019-2022 PureStake Inc.
+// This file is part of Moonbeam.
+
+// Moonbeam is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Moonbeam is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Moonbeam.  If not, see <http://www.gnu.org/licenses/>.
+
+//! Primitives for VRFs
+use nimbus_primitives::NimbusId;
+
+/// A Trait to lookup keys from AuthorIds
+pub trait KeysLookup<Keys> {
```

Comment:
I think it's better to add a second generic `AuthorId` (that will be instanciated `NimbusId` in moonbeam case), but it's more clean and generic.

Reply:
Sure, I'll do it. I also prefer traits generic over inputs, but that is not how nimbus_primitives::AccountLookup is written.